### PR TITLE
Enhance/79 responsive for blogs and blog individual page

### PR DESCRIPTION
**Pull request by Trent1900**
### Descriptions
enchance the responsive design for both blog and single blog page, and update necessary test.
- Link to UI design (if possible):
https://www.figma.com/file/q2ZHM8s0LcNxz3otGFLDzO/Website?node-id=101-3&t=f4tBDpdQJTAMmjuw-0
### Provide screenshots of no warning, or error in the console.
image

https://user-images.githubusercontent.com/63524682/225511821-2da5ea43-48af-4c4b-9d6d-d9f348101763.mp4

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: __tests__/pages/blog/[id].test.tsx [hunk 1]
```diff
@@ -72,4 +73,49 @@ describe("Blog single page", () => {
...
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // Deprecated
+        removeListener: jest.fn(), // Deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+    component = render(<Blog />);
```
Comment (Trent1900) [inline __tests__/pages/blog/[id].test.tsx:92]: 这里在模拟window.matchMedia.maches,让他返回true。

File: __tests__/pages/blog/[id].test.tsx [hunk 2]
```diff
@@ -72,4 +73,49 @@ describe("Blog single page", () => {
...
+        matches: false,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(), // Deprecated
+        removeListener: jest.fn(), // Deprecated
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
```
Comment (Trent1900) [inline __tests__/pages/blog/[id].test.tsx:114]: 这里在模拟window.matchMedia.maches,让他返回false。

File: __tests__/pages/blog/[id].test.tsx
Comment (Trent1900) [inline __tests__/pages/blog/[id].test.tsx:1]: The reason why I have to modify the test file, is that the test branch coverage is 50%.

**Review by yunshen513 — APPROVED** [review]
LGTM

**Review by huchshm — APPROVED** [review]

**Pull request merged**